pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,198 @@
+		q.Add(req)
+	}
+}
+
+// Update implements EventHandler
+func (e *EnqueueRequestForAnnotation) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.MetaOld); ok {
+		q.Add(req)
+	} else if ok, req := e.getAnnotationRequests(evt.MetaNew); ok {
+		q.Add(req)
+	}
+}
+
+// Delete implements EventHandler
+func (e *EnqueueRequestForAnnotation) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// Generic implements EventHandler
+func (e *EnqueueRequestForAnnotation) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// getAnnotationRequests will check if the object has the annotations for the watch handler and requeue
+func (e *EnqueueRequestForAnnotation) getAnnotationRequests(object metav1.Object) (bool, reconcile.Request) {
+	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type.String() {
+		namespacedNameString, ok := object.GetAnnotations()[NamespacedNameAnnotation]
+		if !ok {
+			log.Info("Unable to find the annotation for handle watch annotation",
+				"resource", object, "annotation", NamespacedNameAnnotation)
+		}
+		if len(namespacedNameString) < 1 {
+			return false, reconcile.Request{}
+		}
+		return true, reconcile.Request{NamespacedName: parseNamespacedName(namespacedNameString)}
+	}
+	return false, reconcile.Request{}
+}
+
+// parseNamespacedName will parse the value informed in the NamespacedNameAnnotation and return types.NamespacedName
+// with. Note that if just one value is informed, then it will be set as the name.
+func parseNamespacedName(namespacedNameString string) types.NamespacedName {
+	values := strings.SplitN(namespacedNameString, "/", 2)
+	if len(values) == 1 {
+		return types.NamespacedName{
+			Name:      values[0],
+			Namespace: "",
+		}
+	}
+	if len(values) >= 2 {
+		return types.NamespacedName{
+			Name:      values[1],
+			Namespace: values[0],
+		}
+	}
+	return types.NamespacedName{}
+}
+
+// SetWatchOwnerAnnotation is a helper method to add the watch annotation-based with the owner NamespacedName and the
+// schema.GroupKind to the object provided. This allows you to declare the watch annotations of an owner to an object.
+// If a annotation to the same object already exists, it'll be overwritten with the newly provided version.
+func SetWatchOwnerAnnotation(owner, object metav1.Object, ownerGK schema.GroupKind) error {
+
+	if len(owner.GetName()) < 1 {
+		return fmt.Errorf("%T has not a name, cannot call SetWatchOwnerAnnotation", owner)
+	}
+
+	if len(ownerGK.Kind) < 1 {
+		return fmt.Errorf("Owner Kind is not found, cannot call SetWatchOwnerAnnotation")
+	}
+
+	if len(ownerGK.Group) < 1 {
+		return fmt.Errorf("Owner Group is not found, cannot call SetWatchOwnerAnnotation")
+	}
+
+	object.SetAnnotations(map[string]string{
```

Comment:
I think this needs to be 

```golang
annotations := object.GetAnnotations()
annotations[NamespacedNameAnnotation] = fmt.Sprintf("%v/%v", owner.GetNamespace(), owner.GetName())
annotations[TypeAnnotation] = ownerGK.String()
object.SetAnnotations(annotations)
```

or else you'll drop any other set annotations on `object` when using `SetWatchOwnerAnnotation`

Reply:
Oh .. really tks for the good catcher .. it was a silly mistake ..
fixed and added tests to ensure as well.